<?xml version="1.0" encoding="UTF-8"?>

<!DOCTYPE beans PUBLIC "-//SPRING//DTD BEAN//EN" "http://www.springframework.org/dtd/spring-beans.dtd">

<beans>
    
    <!-- Ajax -->
    
    <bean id="dummyHandler" class="org.springmodules.xt.test.ajax.DummyHandler"/>
    
    <bean id="dummySubmitHandler" class="org.springmodules.xt.test.ajax.DummySubmitHandler"/>
    
    <bean id="ajaxValidationHandler" class="org.springmodules.xt.ajax.validation.DefaultValidationHandler"/>
    
    <bean id="redirectExResolver1" class="org.springmodules.xt.ajax.support.RedirectExceptionResolver">
        <property name="redirectUrl" value="/test/redirect1.html"/>
    </bean>
    
    <bean id="redirectExResolver2" class="org.springmodules.xt.ajax.support.RedirectExceptionResolver">
        <property name="redirectUrl" value="/test/redirect2.html"/>
    </bean>
    
    <bean id="ajaxInterceptor" class="org.springmodules.xt.ajax.AjaxInterceptor">
        <property name="handlerMappings">
            <props>
                <prop key="/ajax/test.action">dummyHandler</prop>
                
                <prop key="/ajax/submit.*">ajaxValidationHandler</prop>
                <prop key="/ajax/submit.action">dummySubmitHandler</prop>
                
                <prop key="/ajax/chainedSubmit.action">dummySubmitHandler, ajaxValidationHandler</prop>
            </props>
        </property>
    </bean>
    
    <bean id="ajaxExceptionHandler" class="org.springmodules.xt.ajax.AjaxExceptionHandler">
        <property name="exceptionMappings">
            <map>
                <entry key="org.springmodules.xt.ajax.support.UnsupportedEventException" value-ref="redirectExResolver1"/>
                <entry key="org.springmodules.xt.ajax.support.EventHandlingException" value-ref="redirectExResolver2"/>
            </map>
        </property>
    </bean>
    
    <!-- -->
    
</beans>